node/src/db_config/persistent_configuration.rs

```diff
@@ -415,19 +415,25 @@ impl PersistentConfiguration for PersistentConfigurationReal {
         self.simple_set_method("start_block", value)
     }
 
-    fn max_block_count(&self) -> Result<u64, PersistentConfigError> {
+    fn max_block_count(&self) -> Result<Option<u64>, PersistentConfigError> {
         match self.get("max_block_count") {
             Ok(max_block_count) => match decode_u64(max_block_count) {
-                Ok(Some(mbc)) => Ok(mbc),
+                Ok(mbc_opt) => Ok(mbc_opt),
+                Err(TypedConfigLayerError::BadNumberFormat(value)) if value.is_empty() => Ok(None),
                 Err(e) => Err(PersistentConfigError::from(e)),
-                Ok(None) => Err(PersistentConfigError::NotPresent),
             },
             Err(e) => Err(PersistentConfigError::from(e)),
         }
     }
 
-    fn set_max_block_count(&mut self, value: u64) -> Result<(), PersistentConfigError> {
-        self.simple_set_method("max_block_count", value)
+    fn set_max_block_count(&mut self, value: Option<u64>) -> Result<(), PersistentConfigError> {
+        self.simple_set_method(
+            "max_block_count",
+            match encode_u64(value) {
+                Ok(Some(mbc)) => mbc,
+                _ => "".to_string(),
+            },
```

Comment:
This should also be done differently. If you have a test for this already, it's going to be easy though. I believe you should simply pass the value in without using a match in the place. The apparatus of ConfigDao will know the way to handle it. Just write:

```
  self.simple_set_method( "max_block_count", value)
 ```

Reply:
I wish it worked like you wrote it; it would be so simple (sorry for the terrible formatting):

```stdout
error[E0277]: `std::option::Option<u64>` doesn't implement `std::fmt::Display`
   --> src/db_config/persistent_configuration.rs:429:51
    |
429 |         self.simple_set_method("max_block_count", value)
    |              -----------------                    ^^^^^ `std::option::Option<u64>` cannot be formatted with the default formatter
    |              |
    |              required by a bound introduced by this call
    |
    = help: the trait `std::fmt::Display` is not implemented for `std::option::Option<u64>`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
note: required by a bound in `persistent_configuration::PersistentConfigurationReal::simple_set_method`
   --> src/db_config/persistent_configuration.rs:559:29
    |
559 |     fn simple_set_method<T: Display>(
    |                             ^^^^^^^ required by this bound in `persistent_configuration::PersistentConfigurationReal::simple_set_method`
```